src/WebJobs.Script/OutOfProc/Rpc/RpcInitializationService.cs

```diff
@@ -161,6 +165,40 @@ private Task InitializeWebHostRuntimeChannelsAsync()
             return Task.CompletedTask;
         }
 
+        private async Task DrainBuffersWithTimeout(TimeSpan timeout)
+        {
+            var workerChannels = _webHostlanguageWorkerChannelManager.GetChannels(_environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName));
+            Task delayTask = Task.Delay(timeout);
+            if (workerChannels != null)
+            {
+                foreach (string workerId in workerChannels.Keys)
+                {
+                    await Task.WhenAny(delayTask, DrainBuffer(workerChannels[workerId]?.Task.Result, timeout));
+                }
+            }
+        }
+
+        private Task DrainBuffer(ILanguageWorkerChannel languageWorkerChannel, TimeSpan timeout)
+        {
+            using (ManualResetEventSlim mres1 = new ManualResetEventSlim(false))
+            {
+                IDisposable subscription = Observable
+                                    .Interval(TimeSpan.FromSeconds(1.0))
```

Comment:
waiting 1 second is too long. Please check if there is a way to avoid the wait before checking if buffer is drained. For example, check buffer after an invocation completes?